notes/general/intro.md

```diff
@@ -0,0 +1,32 @@
+## Introduction
+
+### What is Algorithm?
+
+Informally, algorithms is well-defined sequence of steps that takes some values (input) and gives out some values after calculation (output).
+
+The words 'algorithm' and 'algorism' come from the name al-Khwārizmī. Al-Khwārizmī was a Persian mathematician, astronomer, geographer, and scholar. 
+
+### Example
+
+One of problem of sorting a set of elements in ascending or descending order can be defined formally as an algorithm as follows:
+
+_Input_ : A sequence of n numbers <a1,a2,...an>
```

Comment:
```A sequence of _n_ numbers {a<sub>1</sub>, a<sub>2</sub>, ... a<sub>n</sub>}``` will output:

A sequence of _n_ numbers {a<sub>1</sub>, a<sub>2</sub>, ... a<sub>n</sub>}